content/blog/jetstream-java-client-multi-tool.md

```diff
@@ -0,0 +1,55 @@
++++
+categories = ["Clients", "Java", "JetStream"]
+date = "2021-05-10"
+tags = ["java", "stream", "publish", "subscribe", "benchmark", "tool"]
+title = "JetStream Multi Tool"
+author = "Scott Fauerbach"
++++
+
+As we were building the [NATS Java client](https://github.com/nats-io/nats.java) we realized that we wanted code to run the client through its paces.
+We built unit tests, example code and added basic functionality to our existing benchmarking tool.
+
+As we talked to prospective users, they were asking for benchmarks across the variety of different ways we can publish and subscribe.
+For instance see publishing synchronous versus asynchronous. 
+Push versus pull consuming. Different amounts of threading with shared or individual connections to the server.
+You get the idea. So we set out to build a tool to do just that.
+
+No matter the functionality being tested, there were several things in common:
+
+* Threading and connections to the server.
+* The ability to be run with command line arguments or from java code.
+* Statistics.
+* Commonality between varieties of publishing like payload size.
+* Commonality between varieties of subscriptions like ack policy.
+
+So it made sense to put everything into one program and the 
+[JetStream Multi Tool](https://github.com/nats-io/nats.java/tree/main/src/examples/java/io/nats/examples/jsmulti)
```

Comment:
Could we add Java Client to the name -> NATS Java Client JetStream Multi Tool? I want to avoid any confusion with the NATS CLI

Reply:
Fixed and added additional clarifying comments